### PR TITLE
Enabled movable Custom Views

### PR DESCRIPTION
**Pull request by sbatten**
This PR refs #85164

- Updates the tree view to enable updating of the styling
- Add events to ViewDescriptorService that allow listening for Container and Location change
- Add methods to ViewDescriptorService to get the view location. This works even before the container is registered and will trigger an event if the cached data turns out to be stale.
- Outline follows similar logic so included as well

### CODE REVIEW

File: src/vs/workbench/services/views/browser/viewDescriptorService.ts
Comment (sandy081) [inline src/vs/workbench/services/views/browser/viewDescriptorService.ts]: I think we shall also check location change here even though currently location change is not yet supported in  registry. This should be ready for future if it is allowed.

I would recommend to have a single method that can be reused when views are moved in registry or in cache.

```ts
this.removeViews(from, views);
 this.addViews(to, views);

  const oldLocation = this.viewContainersRegistry.getViewContainerLocation(from)!;
 const newLocation = this.viewContainersRegistry.getViewContainerLocation(to)!;

  if (oldLocation !== newLocation) {
 	this._onDidChangeLocation.fire({ views, from: oldLocation, to: newLocation });
 }

 this._onDidChangeContainer.fire({ views, from, to });
```

**Review by sandy081 — APPROVED** [review]
Reviewed changes in `ViewDescriptorService`. Requested one change.
Approving it expecting that requested change is taken care of.

Thanks.

File: src/vs/workbench/services/views/browser/viewDescriptorService.ts
Comment (sbatten) [reply src/vs/workbench/services/views/browser/viewDescriptorService.ts]: @sandy081 thanks, will take care of this.

**Comment by sbatten** [comment]
It would still be great to get approval from:
@joaomoreno on `TreeView` styling changes
@alexr00 for general `CustomView` updates

**Review by joaomoreno — APPROVED** [review]
Changes in `listService` make sense!

**Review by alexr00 — APPROVED** [review]
The custom view changes look good!

**Comment by sandy081** [comment]
@sbatten Merging the PR as we have all approvals. 
I will also take care of doing the change I suggested after merging.

**Pull request merged**
